powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/TemporaryKeyBehavior.java

```diff
@@ -0,0 +1,345 @@
+        private PrivateKey privateKey;
+        private PublicKey publicKey;
+    }
+
+    private final LocalizationProvider localizationProvider;
+    private final ApplicationVersionRepository applicationVersionRepository;
+    private final ActivationRepository activationRepository;
+    private final MasterKeyPairRepository masterKeyPairRepository;
+    private final TemporaryKeyRepository temporaryKeyRepository;
+    private final TemporaryPrivateKeyConverter temporaryPrivateKeyConverter;
+    private final ServerPrivateKeyConverter serverPrivateKeyConverter;
+
+    private final KeyConvertor keyConvertor = new KeyConvertor();
+    private final KeyGenerator keyGenerator = new KeyGenerator();
+    private final PowerAuthServerKeyFactory keyFactory = new PowerAuthServerKeyFactory();
+
+    @Autowired
+    public TemporaryKeyBehavior(LocalizationProvider localizationProvider, ApplicationVersionRepository applicationVersionRepository, ActivationRepository activationRepository, MasterKeyPairRepository masterKeyPairRepository, TemporaryKeyRepository temporaryKeyRepository, TemporaryPrivateKeyConverter temporaryPrivateKeyConverter, ServerPrivateKeyConverter serverPrivateKeyConverter) {
+        this.localizationProvider = localizationProvider;
+        this.applicationVersionRepository = applicationVersionRepository;
+        this.activationRepository = activationRepository;
+        this.masterKeyPairRepository = masterKeyPairRepository;
+        this.temporaryKeyRepository = temporaryKeyRepository;
+        this.temporaryPrivateKeyConverter = temporaryPrivateKeyConverter;
+        this.serverPrivateKeyConverter = serverPrivateKeyConverter;
+    }
+
+    @Transactional
+    public TemporaryPublicKeyResponse requestTemporaryKey(TemporaryPublicKeyRequest request) throws GenericServiceException, InvalidKeySpecException, CryptoProviderException, GenericCryptoException, InvalidKeyException {
+
+        // Get claims from JWT
+        final DecodedJWT decodedJWT = JWT.decode(request.getJwt());
+        final TemporaryPublicKeyRequestClaims requestClaims = buildTemporaryKeyClaims(decodedJWT);
+
+        // Validate claims
+        final String error = validateDecodedClaims(requestClaims);
+        if (error != null) {
+            throw new GenericServiceException("INVALID_TEMPORARY_KEY_REQUEST", error);
+        }
+
+        // Obtain verifier secret and check JWT signature
+        final TemporaryKeyResult temporaryKeyResult = obtainTemporaryKeyResult(requestClaims);
+        try {
+            final Algorithm algorithm = Algorithm.HMAC256(temporaryKeyResult.getSecretKeyBytes());
+            final JWTVerifier verifier = JWT.require(algorithm).build();
+            verifier.verify(decodedJWT);
+        } catch (JWTVerificationException ex){
+            logger.debug("JWT token verification failed: {}", ex.getMessage(), ex);
+            throw new GenericServiceException("INVALID_TEMPORARY_KEY_REQUEST", "Invalid temporary key request");
```

Comment:
The error is not in `ServiceError` class, is this OK? Usually the pattern is e.g.:

```
throw localizationProvider.buildExceptionForCode(ServiceError.INVALID_REQUEST);
```